lrlex/src/lib/lexer.rs

```diff
@@ -125,6 +126,19 @@ impl<StorageT> Rule<StorageT> {
             target_state,
         })
     }
+
+    /// Return this rule's token ID, if any.
+    ///
+    /// If `Some`, this specifies the ID that lexemes resulting from this rule will have. If
+    /// `None`, then this rule specifies lexemes which should not appear in the user's input.
+    pub fn tok_id(&self) -> Option<StorageT> {
+        self.tok_id
```

Comment:
It is possible I'm missing something obvious, but I'm wondering if there is a reason that this doesn't return a `cfgrammar::TIdx` or some such?

Reply:
I also wondered that, but it's currently storing a `StorageT`. I can't fully remember what I might expect this to be, to be honest...

Reply:
Looks like it is also exposed publicly as `StorageT` via the [`lrpar::Lexeme`](https://docs.rs/lrpar/latest/lrpar/trait.Lexeme.html#tymethod.tok_id) trait [here](https://github.com/softdevteam/grmtools/blob/master/lrlex/src/lib/lexer.rs#L396).  So it doesn't seem to be unprecedented.

Reply:
Would it be better as `TIdx`? If so, we might as well make the API as unsucky as we can reasonably make it!

Reply:
Looking further into it, I don't think `TIdx` would actually the right thing it should correspond to.  For example, [this](https://github.com/softdevteam/grmtools/blob/master/cfgrammar/src/lib/yacc/grammar.rs#L488-L492) usage in `token_name` isn't referring to one of these `tok_id` values at all, but an index into a `Grammar` specific array.

So, I'm leaning towards it being okay as is, at the very least nothing is ringing any bells for it being any of the other opaque type wrappers, and it certainly doesn't appear that `TIdx` would be right.

Reply:
Excellent detective work! OK I feel adequately comfortable with this PR now then.